CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 3.11)
+project(Venoge VERSION 0.0.0 LANGUAGES CXX)
+
+# Get the correct target locations among all platforms
+include(GNUInstallDirs)
+
+# Target C++11 for now
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# 3rd parties and dependancies
+find_package(Eigen3 3.3 QUIET REQUIRED)
+
+# Make tests optional
+option(BUILD_TESTS "Build test targets" OFF)
```

Comment:
Small detail: I like to put the `option`s at the top of the CMakeLists, because they are somehow "arguments" to cmake, and I usually expect to find them there.

Reply:
I follow a slightly different logic, where I set up the stage then customize behaviors via options. Let's hear also from the other guys :)

Reply:
I do not have a strong opinion regarding it's location. However, would it make sense to have the tests built by default?

Reply:
They will be on by default once I am sure of the setup and that it works (see the other comment in the test file).